### DISCUSSION TITLE
RavanaAPMS v1.0.0

### DISCUSSION THREAD

**Discussion by dulanjayabhanu** [RAVANA-Final-Project/RavanaAPMS#7] [Ideas]
## 🎉 Ravana Adventure Park Management System v1.0.0 🎉

We are thrilled to announce the first official release of the Ravana Adventure Park Management System! This version marks the culmination of our efforts to provide a comprehensive and user-friendly application for managing the various aspects of our adventure park operations.

### What's New 🚀

- **👤 User Management**: Efficiently manage users, roles, and permissions.
- **📅 Activity Management**: Schedule and manage adventure activities and leadership programs.
- **🛠️ Equipment Management**: Track equipment inventory, maintenance, and usage.
- **🛍️ Order and Payment Management**: Handle orders, offers, and payments with invoice generation.
- **👔 Employee Management**: Manage employee details, job roles, and attendance.
- **📊 Report Generation**: Generate extensive reports to analyze different aspects of park operations.
- **🔒 Secure Access**: Implement role-based access control for data security and integrity.

### Key Features ✨

- **Admin 👨‍💼**
  - Secure Login/Logout
  - User Registration and Management
  - Order Management (Add, Update, Delete Orders, Manage Offers)
  - Activity Management (Add, Update, Inactivate Activities)
  - Payment Handling and Invoice Generation
  - Employee Management (Add, Update, Inactivate Employees, Manage Job Roles)
  - Comprehensive Reporting (Income, Employee, Customer Reports)

- **Super Admin 👑**
  - All Admin Privileges (excluding managing users, activities, and employees)
  - Admin Management
  - Equipment Manager Management
  - Report Viewing and Generation (Invoices, Orders, Income, Attendance, Employee, Customer, Equipment, Outdated Equipment)

- **Equipment Manager 🔧**
  - Equipment Registration and Management
  - Brand Management
  - Stock Management
  - Monthly Outdated Equipment Reports

### Installation Guide ⚙️

Before you begin, ensure your PC has MySQL Database Management System installed. You can download MySQL from [here](https://dev.mysql.com/downloads/installer/).

1. **Download the setup file from the releases page:**
    [Download Setup File](https://github.com/yourusername/ravana-adventure-park/releases)

2. **Run the setup file to install the application directly.**

3. **Download the SQL backup file from the repository:**
    [Download SQL Backup File](https://github.com/yourusername/ravana-adventure-park/releases)

4. **Restore the SQL backup file:**
    Open your MySQL client and run the following command to restore the database:
    ```sql
    mysql -u yourusername -p yourdatabase < path/to/your/sqlbackupfile.sql
    ```

Thank you for using the Ravana Adventure Park Management System! 🌟

## What's Changed
* Super_Admin_Attendance_Bug_Fixed by @Kusal-Gunawardana-KMPG in https://github.com/RAVANA-Final-Project/RavanaAPMS/pull/1
* Important Bug Fixes Regarding Manage Activities by @Karannagoda in https://github.com/RAVANA-Final-Project/RavanaAPMS/pull/2
* Implements the assets pack by @Kusal-Gunawardana-KMPG in https://github.com/RAVANA-Final-Project/RavanaAPMS/pull/4
* Verify merging to the master Branch by @dulanjayabhanu in https://github.com/RAVANA-Final-Project/RavanaAPMS/pull/5
* Final Conflict Fixing by @Kusal-Gunawardana-KMPG in https://github.com/RAVANA-Final-Project/RavanaAPMS/pull/6

## New Contributors
* @Kusal-Gunawardana-KMPG made their first contribution in https://github.com/RAVANA-Final-Project/RavanaAPMS/pull/1
* @Karannagoda made their first contribution in https://github.com/RAVANA-Final-Project/RavanaAPMS/pull/2
* @dulanjayabhanu made their first contribution in https://github.com/RAVANA-Final-Project/RavanaAPMS/pull/5

**Full Changelog**: https://github.com/RAVANA-Final-Project/RavanaAPMS/commits/v1.0.0

<hr /><em>This discussion was created from the release <a href='https://github.com/RAVANA-Final-Project/RavanaAPMS/releases/tag/v1.0.0'>RavanaAPMS v1.0.0</a>.</em>

**Comment by iamvirul** [comment]
I have a question. In your project readme, you guys mention using "npm install" to download dependencies. I believe this may be a mistake.

![image](https://github.com/user-attachments/assets/e94d9ef9-3e1b-4edb-babe-5bb0619f0538)

**Reply by dulanjayabhanu** [reply]
**Thank you so much for bringing this to our attention! ⚡**

We've carefully reviewed your feedback and have corrected the incorrect installation steps in our guide. 🍵 Additionally, we've added a detailed project setup guide for both `NetBeans` and `Eclipse` to ensure a smoother experience for all developers. 🛠️

Your input has been invaluable in helping us improve the project documentation, and we truly appreciate your support. If you have any further suggestions or encounter any other issues, please don't hesitate to let us know! 🚀
